dlink-admin/src/main/java/com/dlink/service/impl/DataBaseServiceImpl.java

```diff
@@ -75,8 +75,11 @@ public boolean saveOrUpdateDataBase(DataBase dataBase) {
             return false;
         }
         if (Asserts.isNull(dataBase.getId())) {
-            checkHeartBeat(dataBase);
-            return save(dataBase);
+            try {
+                checkHeartBeat(dataBase);
```

Comment:
there is a simpler scheme,you can save first before checkheart, checkheart exception maybe useful, could log it. finally return save function result.

Reply:
There is a problem with processing according to the replacement execution order. Because the time of saving data and exception throwing is too short, the front end displays only an exception popup, and does not give the user a prompt whether to save successfully. But I think what you said about keeping exception information is also correct. Can I print the exception after exception capture and save it in the finally module, so that the prompt of successful front-end saving still exists, and the exception information can also be found in the log.

Reply:
> there is a simpler scheme,you can save first before checkheart, checkheart exception maybe useful, could log it. finally return save function result.

There is a problem with processing according to the replacement execution order. Because the time of saving data and exception throwing is too short, the front end displays only an exception popup, and does not give the user a prompt whether to save successfully. But I think what you said about keeping exception information is also correct. Can I print the exception after exception capture and save it in the finally module, so that the prompt of successful front-end saving still exists, and the exception information can also be found in the log.

Reply:
There is another way to return the saved results and heartbeat detection results to the controller layer, and make judgments in the controller layer to throw the saved results and exception results together to the front end, but this is not elegant in the lmpl layer. Do you have any good suggestions? @aiwenmo @leechor

Reply:
![企业微信截图_16735903121196](https://user-images.githubusercontent.com/44955013/212250082-108f744c-8099-4a9f-b888-47ae1572a1c9.png)
Is this a better way to deal with it?

Reply:
dinky is refracting now, I think you could redesign and improve this feature logic.